src/cache-restore.ts

```diff
@@ -29,7 +29,8 @@ export const restoreCache = async (
     );
   }
 
-  const primaryKey = `setup-go-${platform}-go-${versionSpec}-${fileHash}`;
+  const cacheKeyPrefix = core.getInput('cache-key-prefix') || '';
+  const primaryKey = `${cacheKeyPrefix}setup-go-${platform}-go-${versionSpec}-${fileHash}`;
```

Comment:
from @dsame [comment](https://github.com/actions/setup-go/pull/366/files/acb2fd804097c1dffc2eb1075b3696da4b4bb27c#r1212666065)
```suggestion
  const cacheKeyPrefix = core.getInput('cache-key-prefix') || 'setup-go'
  const primaryKey = `${cacheKeyPrefix}-${platform}-go-${versionSpec}-${fileHash}`;
```